### PR TITLE
Fix workspace selection persistence and delete repair

### LINKED ISSUE #22: Workspace selection and deletion need active-workspace persistence and repair
Summary
Workspace selection does not appear to persist robustly, and deleting the current workspace can crash the app. The current iOS selector switches the in-memory active workspace, but it does not repair or persist the active-workspace identifier that launch restore depends on.

Likely cause
`WorkspaceSelectorView` only calls `windowManager.setActiveWorkspace(workspace)` when a row is tapped, and the delete helpers just delete the model row through `WorkspaceStore`. `WindowManager.setActiveWorkspace` only swaps the in-memory workspace and refreshes windows, while app launch restore depends on `SettingsStore.activeWorkspaceId`.

Relevant iOS references
- `WorkspaceSelectorView.swift` lines 187-220 and 284-334
- `WindowManager.swift` lines 78-108
- `AndBibleApp.swift` lines 700-725 where active workspace restore is performed from `SettingsStore.activeWorkspaceId`
- `SettingsStore.swift` lines 223-234 for the persisted active-workspace key

Android comparison
Android's workspace selector returns the selected workspace ID to the host activity. `WorkspaceSelectorActivity.kt` uses `goToWorkspace(...)` around lines 578-603, and `MainBibleActivity.kt` handles `WORKSPACE_CHANGED` at lines 1847-1858 to update the current workspace ID explicitly. That keeps the selection contract and restore path in one place.

Expected
Persist the selected workspace when switching, and define a safe repair path when the active workspace is removed instead of leaving the app to fall back implicitly or crash.

### PR DESCRIPTION
**Pull request by primetheus**
## Summary

This fixes workspace selection so switching workspaces updates both the live `WindowManager` state and the persisted active workspace used during launch restore. It also allows deleting the active workspace when another workspace survives by repairing selection before the delete happens, matching the Android behavior.

## Scope

- Adds a `WorkspaceSelectionService` in `BibleCore` to coordinate `WorkspaceStore`, `SettingsStore`, and `WindowManager`.
- Updates `WorkspaceSelectorView` to use that service for selecting, creating, and deleting workspaces.
- Keeps final-workspace deletion blocked, but no longer blocks active-workspace deletion when a valid survivor exists.
- Adds deterministic service-level tests instead of UI tests so the regression coverage does not depend on SwiftUI timing or accessibility queries.

## Contract References

- Issue: Closes #22
- Android parity: Android persists the selected workspace ID when switching and repairs to a surviving workspace before deleting the active workspace.
- Test strategy: model/service-level coverage for persistence and repair behavior, avoiding flaky UI selector assertions.

## Change Details

- `WorkspaceSelectionService.activate(_:)` calls `WindowManager.setActiveWorkspace` and writes `SettingsStore.activeWorkspaceId` together.
- `WorkspaceSelectionService.deleteWorkspaces(_:)` refuses to delete all persisted workspaces, switches to the first survivor when the active workspace is being deleted, deletes the requested persisted workspaces, then repairs persisted active state.
- `WorkspaceSelectorView` now routes selection, creation, single delete, and multi-delete through the service.
- Added regression tests for selected-workspace persistence, active delete repair, inactive delete preservation, batch delete repair, and final-workspace delete refusal.

## Validation Evidence

- `git diff --check`
- Focused workspace tests on iPhone 17 simulator: 6 passed, 0 failed
- `xcodebuild test -project AndBible.xcodeproj -scheme AndBible -destination 'platform=iOS Simulator,name=iPhone 17' -only-testing:AndBibleTests CODE_SIGNING_ALLOWED=NO`: 180 passed, 0 failed
- Installed and manually verified the branch build on booted iPhone 17 and iPad Pro 11-inch simulators

## Risks / Follow-ups

- I kept the fix out of UI tests because the behavior is synchronous state coordination and UI-level XCTest has been the flaky layer.
- No known follow-up is required for #22 after this fix, unless additional workspace restore edge cases are reported.

### CODE REVIEW

File: Sources/BibleCore/Sources/BibleCore/Services/WorkspaceSelectionService.swift [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        }
+
+        let activeWorkspaceID = windowManager.activeWorkspace?.id ?? settingsStore.activeWorkspaceId
+        if let activeWorkspaceID, deletableIDs.contains(activeWorkspaceID) {
+            activate(survivingWorkspaces[0])
+        }
+
+        for workspace in persistedWorkspaces where deletableIDs.contains(workspace.id) {
+            workspaceStore.delete(workspace)
+        }
```
[the remark this answers is not in the document]
Comment (primetheus) [reply Sources/BibleCore/Sources/BibleCore/Services/WorkspaceSelectionService.swift]: Good catch. I changed this to use a new `WorkspaceStore.deleteWorkspaces(_:)` helper so multi-delete marks the selected workspaces for deletion and saves the SwiftData context once, after the active-workspace survivor has already been selected. I also kept the existing single-delete path unchanged.

I validated this with the focused workspace/store tests and the full `AndBibleTests` target locally before pushing.
